### PR TITLE
Use functional setState when derived from previous

### PR DESCRIPTION
**Pull request by munyari**
Associated Issue:  #5512

### Summary of Changes

* Use functional setState when derived from previous

### Test Plan

[ ] Verify that each component still works as expected (I'll flesh this out in the morning)

Here's the Debugger's Testing doc
https://docs.google.com/document/d/1oBMRxV8A2ag2t22YsQOxTdEv0mXKzIg0tggJjRkU1S0/edit#.
Feel free to improve it!

### CODE REVIEW

File: src/components/shared/Dropdown.js [hunk 1]
```diff
@@ -28,9 +28,9 @@ export class Dropdown extends Component<Props, State> {
   }
 
   toggleDropdown = (e: SyntheticKeyboardEvent<HTMLElement>) => {
-    this.setState({
-      dropdownShown: !this.state.dropdownShown
-    });
+    this.setState(prevState => ({
+      dropdownShown: !prevState.dropdownShown;
```
Comment (nyrosmith) [inline src/components/shared/Dropdown.js]: The semicolon causes a Syntax error

**Comment by nyrosmith** [comment]
Are this all places where we rely on previous state?

**Comment by munyari** [comment]
@nyrosmith added one more instance. I think that is all of them (verified by visual inspection)

**Comment by nyrosmith** [comment]
Great work!
Thank you @munyari

**Pull request merged**